packages/giselle-engine/src/core/generations/generate-text.ts

```diff
@@ -72,6 +75,51 @@ export async function generateText(args: {
 			},
 		}),
 	]);
+
+	const workspaceId = await extractWorkspaceIdFromOrigin({
+		storage: args.context.storage,
+		origin: args.generation.context.origin,
+	});
+
+	const usageLimitStatus = await checkUsageLimits({
+		workspaceId,
+		generation: args.generation,
+		fetchUsageLimitsFn: args.context.fetchUsageLimitsFn,
+	});
+	if (usageLimitStatus.type === "error") {
+		const failedGeneration = {
+			...runningGeneration,
+			status: "failed",
+			failedAt: Date.now(),
+			error: {
+				name: usageLimitStatus.error,
+				message: usageLimitStatus.error,
+				dump: usageLimitStatus,
+			},
+		} satisfies FailedGeneration;
+		await Promise.all([
+			setGeneration({
+				storage: args.context.storage,
+				generation: failedGeneration,
+			}),
+			setNodeGenerationIndex({
+				storage: args.context.storage,
+				nodeId: runningGeneration.context.actionNode.id,
+				origin: runningGeneration.context.origin,
+				nodeGenerationIndex: {
+					id: failedGeneration.id,
+					nodeId: failedGeneration.context.actionNode.id,
+					status: "failed",
+					createdAt: failedGeneration.createdAt,
+					queuedAt: failedGeneration.queuedAt,
+					startedAt: failedGeneration.startedAt,
+					failedAt: failedGeneration.failedAt,
+				},
+			}),
+		]);
+		throw new UsageLimitError(usageLimitStatus.error);
```

Comment:
I'd like to return either 402 or 406 when this error is thrown in the http/router. It doesn't have to be in this pull request.